tensorflow/lite/micro/kernels/xtensa/svdf.cc

```diff
@@ -288,45 +311,81 @@ TfLiteStatus Eval(TfLiteContext* context, TfLiteNode* node) {
                              weights_time, bias, params, activation_state,
                              output, data);
 #else
+  EvalInt16SvdfReference(context, node, input, weights_feature, weights_time,
+                         bias, params, activation_state, output, data);
+  return kTfLiteOk;
+#endif  // defined(HIFI4) || defined(HIFI5)
+}
+
+TfLiteStatus Eval(TfLiteContext* context, TfLiteNode* node) {
+  auto* params = static_cast<TfLiteSVDFParams*>(node->builtin_data);
+
+  const TfLiteEvalTensor* input =
+      tflite::micro::GetEvalInput(context, node, kSvdfInputTensor);
+  const TfLiteEvalTensor* weights_feature =
+      tflite::micro::GetEvalInput(context, node, kSvdfWeightsFeatureTensor);
+  const TfLiteEvalTensor* weights_time =
+      tflite::micro::GetEvalInput(context, node, kSvdfWeightsTimeTensor);
+  // TODO(tflite-micro#1751) account for optional bias tensor
+  const TfLiteEvalTensor* bias =
+      (NumInputs(node) == 5)
+          ? tflite::micro::GetEvalInput(context, node, kSvdfBiasTensor)
+          : nullptr;
+  TfLiteEvalTensor* activation_state = tflite::micro::GetMutableEvalInput(
+      context, node, kSvdfInputActivationStateTensor);
+  TfLiteEvalTensor* output =
+      tflite::micro::GetEvalOutput(context, node, kSvdfOutputTensor);
+
+  TFLITE_DCHECK(node->user_data != nullptr);
+  const OpDataSvdf& data = *(static_cast<const OpDataSvdf*>(node->user_data));
+
   switch (weights_feature->type) {
     case kTfLiteFloat32: {
       EvalFloatSvdfReference(
           context, node, input, weights_feature, weights_time, bias, params,
           data.scratch_tensor_index, activation_state, output);
-      return kTfLiteOk;
       break;
     }
 
     case kTfLiteInt8: {
       switch (weights_time->type) {
         case kTfLiteInt16: {
+#if defined(HIFIMINI)
+          return EvalIntegerSvdfHifimini(context, node, input, weights_feature,
+                                         weights_time, bias, params,
+                                         activation_state, output, data);
+#elif defined(HIFI4) || defined(HIFI5)
+          return EvalIntegerSvdfHifi(context, node, input, weights_feature,
+                                     weights_time, bias, params,
+                                     activation_state, output, data);
+#else
```

Comment:
let's replace this with a call to `EvalInt8(context, node)`.

That will reduce duplication and improve the test converage for the code in EvalInt8.

Reply:
done